File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Quest.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Quest : ITableOperate
{ private const string TAB_FILE_DATA = "quest.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_NAME,
ID_IS_LOOP,
ID_QUEST_TYPE,
ID_QUEST_VALUE,
ID_BACK_ID,
ID_REWARD_GOLD,
ID_REWARD_DOLLAR,
ID_REWARD_POWER,
ID_REWARD_CHANCE,
ID_REWARD_ID,
ID_ATTITEM_NUM,
ID_HPITEM_NUM,
ID_REWARD_ONE,
ID_REWARD_TWO,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AttitemNUM;
 public int AttitemNUM { get{ return m_AttitemNUM;}}

private int m_BackID;
 public int BackID { get{ return m_BackID;}}

private int m_HpitemNUM;
 public int HpitemNUM { get{ return m_HpitemNUM;}}

private int m_IsLoop;
 public int IsLoop { get{ return m_IsLoop;}}

private string m_Name;
 public string Name { get{ return m_Name;}}

private int m_QuestType;
 public int QuestType { get{ return m_QuestType;}}

private int m_QuestValue;
 public int QuestValue { get{ return m_QuestValue;}}

private int m_RewardID;
 public int RewardID { get{ return m_RewardID;}}

private int m_RewardChance;
 public int RewardChance { get{ return m_RewardChance;}}

private int m_RewardDollar;
 public int RewardDollar { get{ return m_RewardDollar;}}

private int m_RewardGold;
 public int RewardGold { get{ return m_RewardGold;}}

private string m_RewardOne;
 public string RewardOne { get{ return m_RewardOne;}}

private int m_RewardPower;
 public int RewardPower { get{ return m_RewardPower;}}

private string m_RewardTwo;
 public string RewardTwo { get{ return m_RewardTwo;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Quest _values = new Tab_Quest();
 _values.m_AttitemNUM =  Convert.ToInt32(valuesList[(int)_ID.ID_ATTITEM_NUM] as string);
_values.m_BackID =  Convert.ToInt32(valuesList[(int)_ID.ID_BACK_ID] as string);
_values.m_HpitemNUM =  Convert.ToInt32(valuesList[(int)_ID.ID_HPITEM_NUM] as string);
_values.m_IsLoop =  Convert.ToInt32(valuesList[(int)_ID.ID_IS_LOOP] as string);
_values.m_Name =  valuesList[(int)_ID.ID_NAME] as string;
_values.m_QuestType =  Convert.ToInt32(valuesList[(int)_ID.ID_QUEST_TYPE] as string);
_values.m_QuestValue =  Convert.ToInt32(valuesList[(int)_ID.ID_QUEST_VALUE] as string);
_values.m_RewardID =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_ID] as string);
_values.m_RewardChance =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_CHANCE] as string);
_values.m_RewardDollar =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_DOLLAR] as string);
_values.m_RewardGold =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_GOLD] as string);
_values.m_RewardOne =  valuesList[(int)_ID.ID_REWARD_ONE] as string;
_values.m_RewardPower =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_POWER] as string);
_values.m_RewardTwo =  valuesList[(int)_ID.ID_REWARD_TWO] as string;

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Quest.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Quest implements ITableOperate {
 public final static String TAB_FILE_DATA = "/quest.txt";
 public static final int ID_NAME = 0 ;
public static final int ID_IS_LOOP = 1 ;
public static final int ID_QUEST_TYPE = 2 ;
public static final int ID_QUEST_VALUE = 3 ;
public static final int ID_BACK_ID = 4 ;
public static final int ID_REWARD_GOLD = 5 ;
public static final int ID_REWARD_DOLLAR = 6 ;
public static final int ID_REWARD_POWER = 7 ;
public static final int ID_REWARD_CHANCE = 8 ;
public static final int ID_REWARD_ID = 9 ;
public static final int ID_ATTITEM_NUM = 10 ;
public static final int ID_HPITEM_NUM = 11 ;
public static final int ID_REWARD_ONE = 12 ;
public static final int ID_REWARD_TWO = 13 ;
public static final int MAX_RECORD = 14 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AttitemNUM;
 public int GetAttitemNUM() { return m_AttitemNUM;}

private int m_BackID;
 public int GetBackID() { return m_BackID;}

private int m_HpitemNUM;
 public int GetHpitemNUM() { return m_HpitemNUM;}

private int m_IsLoop;
 public int GetIsLoop() { return m_IsLoop;}

private String m_Name;
 public String GetName() { return m_Name;}

private int m_QuestType;
 public int GetQuestType() { return m_QuestType;}

private int m_QuestValue;
 public int GetQuestValue() { return m_QuestValue;}

private int m_RewardID;
 public int GetRewardID() { return m_RewardID;}

private int m_RewardChance;
 public int GetRewardChance() { return m_RewardChance;}

private int m_RewardDollar;
 public int GetRewardDollar() { return m_RewardDollar;}

private int m_RewardGold;
 public int GetRewardGold() { return m_RewardGold;}

private String m_RewardOne;
 public String GetRewardOne() { return m_RewardOne;}

private int m_RewardPower;
 public int GetRewardPower() { return m_RewardPower;}

private String m_RewardTwo;
 public String GetRewardTwo() { return m_RewardTwo;}

public boolean LoadTable(HashMap<Integer, Table_Quest> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Quest> _hash = (HashMap<Integer, Table_Quest> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Quest _values = new Table_Quest();
 _values.m_AttitemNUM =  Integer.parseInt((String)valuesList.get(ID_ATTITEM_NUM));
_values.m_BackID =  Integer.parseInt((String)valuesList.get(ID_BACK_ID));
_values.m_HpitemNUM =  Integer.parseInt((String)valuesList.get(ID_HPITEM_NUM));
_values.m_IsLoop =  Integer.parseInt((String)valuesList.get(ID_IS_LOOP));
_values.m_Name =  (String)valuesList.get(ID_NAME);
_values.m_QuestType =  Integer.parseInt((String)valuesList.get(ID_QUEST_TYPE));
_values.m_QuestValue =  Integer.parseInt((String)valuesList.get(ID_QUEST_VALUE));
_values.m_RewardID =  Integer.parseInt((String)valuesList.get(ID_REWARD_ID));
_values.m_RewardChance =  Integer.parseInt((String)valuesList.get(ID_REWARD_CHANCE));
_values.m_RewardDollar =  Integer.parseInt((String)valuesList.get(ID_REWARD_DOLLAR));
_values.m_RewardGold =  Integer.parseInt((String)valuesList.get(ID_REWARD_GOLD));
_values.m_RewardOne =  (String)valuesList.get(ID_REWARD_ONE);
_values.m_RewardPower =  Integer.parseInt((String)valuesList.get(ID_REWARD_POWER));
_values.m_RewardTwo =  (String)valuesList.get(ID_REWARD_TWO);

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

